feat: operatorStateWithSocket

0x0aa0 opened the pull request:
**Motivation:**

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

**Modifications:**

*Describe the modifications you've done.*

**Result:**

*After your change, what will change.*

0x0aa0 committed:
feat: operatorStateWithSocket

pakim249CAL commented:
This runs into stack too deep errors because of the addition of a string in the OperatorWithSocket struct. Because strings are dynamically sized, a memory reference to this string takes up a stack slot. Maybe this can be fixed with --via-ir compilation, but I believe the better immediate solution is to simply not implement this and instead use what is already available. 

In particular, we can obtain the same results offchain by querying the already existing getOperatorState on the OperatorStateRetriever, and querying a getOperatorSocket function that calls the same function on the SocketRegistry.

So I propose that for this PR, we only just add a getOperatorSocket function to OperatorStateRetriever.sol

pakim249CAL commented:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/428 Proposed fix for stack too deep here

pakim249CAL committed:
refactor: return strings outside of struct to avoid stack too deep (#428)

**Motivation:**

The existing PR has stack too deep in tests. Need to move strings
outside of the nested struct. There are a few long term things that we
can do that I will list here, but I don't think it is necessary
immediately because this is just an off-chain view getter that can
easily be changed out later so it's just for reference.

* For some reason, via-ir compilation is disabled. This can often help
with stack too deep issues, but will require some changes in the code
that would expand the scope of the PR too much.
* We can make a socket a fixed size to fix the issue. Right now, a
socket is an arbitrary length string. To me right now, the way socket is
set is totally opaque so I can't really suggest anything about HOW we
can make it a fixed size yet.

**Modifications:**

Move strings outside the nested struct.

**Result:**

Tests will work!

hopeyen reviewed (approved):
lgtm! thanks
